app/scripts/model/dataProxy.js

```diff
@@ -621,63 +637,60 @@ window.DataManagerForIviz = (function($, _) {
                 };
               }
 
-              _result.groups = {};
-              _result.groups.patient = {};
-              _result.groups.sample = {};
-              _result.groups.group_mapping = {};
-              _result.groups.patient.data = _patientData;
-              _result.groups.patient.hasAttrData = _hasPatientAttrData;
-              _result.groups.sample.data = _sampleData;
-              _result.groups.sample.hasAttrData = _hasSampleAttrData;
-              _result.groups.patient.data_indices = {};
-              _result.groups.sample.data_indices = {};
-              _result.groups.patient.data_indices.patient_id =
-                _patientDataIndicesObj;
-              _result.groups.sample.data_indices.sample_id =
-                _sampleDataIndicesObj;
-              _result.groups.group_mapping.sample = {};
-              _result.groups.group_mapping.patient = {};
-              _result.groups.group_mapping.sample.patient =
-                _samplesToPatientMap;
-              _result.groups.group_mapping.patient.sample =
-                _patientToSampleMap;
+              _result.groups = {
+                group_mapping: {
+                  patient_to_sample: _patientToSampleMap,
+                  sample_to_patient: _samplesToPatientMap,
+                  studyMap: _studyToSampleToPatientMap
+                },
+                patient: {
+                  attr_meta: content.util
+                    .sortByClinicalPriority(_.values(_patientAttributes)),
+                  data: _patientUIDData,
+                  has_attr_data: _hasPatientAttrData
+                },
+                sample: {
+                  attr_meta: content.util
+                    .sortByClinicalPriority(_.values(_sampleAttributes)),
+                  data: _sampleUIDData,
+                  has_attr_data: _hasSampleAttrData
+                }
+              };
 
               $.when(self.getCnaFractionData(),
-                self.getMutationCount(),
-                self.getSampleClinicalData(_.keys(_sampleAttributes)),
```

Comment:
`sampleInitData and patientInitData` clinical data are never used here. And is seems like we are trying to cache data for all clinical attribute with isn't required at this point.